feat: Demo setup

deepeshgarg007 opened the pull request:
Ability to setup demo data on new installations

![Demo Setup](https://github.com/frappe/erpnext/assets/42651287/ba99743f-e8b3-46e3-9016-db3ad8185a63)

#### This will set up some master and transactions 
    
##### Masters
- Item, Item Group
- Customer, Customer Group
- Supplier, Supplier Group

##### Transactions
- Sales Invoice
- Purchase Invoice
- Payment Entry

`no-docs` -  THIS IS INTERNAL TO ERPNext. This can be changed/removed any time.

deepeshgarg007 committed:
feat: Demo data creation

deepeshgarg007 committed:
fix: Create transactions

deepeshgarg007 committed:
feat: Clear demo data

deepeshgarg007 committed:
chore: Add more invoices

deepeshgarg007 committed:
chore: Add payment entry

ankush commented:
Just some notes to check before review

- [x] We should add a test for this. A test that doesn't do anything apart from triggering demo setup on new site and checking that it completes successfully. The last demo-data setup became unusable as it had no tests and schema/validations went out of sync over the years.
- [ ] Data should hydrate reports and graphs with enough variety to give users that "aha" moment of finding _"insights"_ from their data.
- [ ] Where will this option to add/remove demo data be present, few options:
    - [x] setup wizard - easily discoverable
    - [ ] first onboarding task in home onboarding widget - slightly less discoverable
- [x] Erasure of demo data: company, transactions, masters.
- [ ] "discoverability" tours using demo data setup.

deepeshgarg007 committed:
test: Add basic test for demo data

deepeshgarg007 committed:
chore: Do not update shopping cart settings on install

deepeshgarg007 committed:
fix: Add demo setup check in setup wizard

deepeshgarg007 committed:
Merge branch 'develop' of https://github.com/frappe/erpnext into demo_data_on_install

deepeshgarg007 committed:
fix: Button to clear demo data

deepeshgarg007 committed:
test: Update demo setup test

deepeshgarg007 committed:
chore: Update records

deepeshgarg007 committed:
fix: reset default company

deepeshgarg007 committed:
Merge branch 'develop' of https://github.com/frappe/erpnext into demo_data_on_install

deepeshgarg007 committed:
fix: Add journal entry in demo

git-avc commented:
Hi:

Maybe you can consider to insert business data using the current year, instead fixed dates. This way, this feature will no get "outdated" on reports, etc ...

ankush committed:
Merge branch 'develop' into demo_data_on_install

ankush commented on erpnext/public/build.json:
```diff
@@ -19,6 +19,7 @@
 		"public/js/queries.js",
 		"public/js/sms_manager.js",
 		"public/js/utils/party.js",
+		"public/js/utils/demo.js",
```
This file is dead actually.

ankush committed:
chore: remove old build system file

[skip ci]

ankush committed:
fix: misc fixes in demo data

- Generate demo data field copy
- absolute imports
- remove whitelisting where it's not required
- telemetry
- banner copy
- move to background
- clear bootinfo after setup

ankush committed:
fix: enqueue at front to speed up demo

ankush commented:
- [ ] hydrate buying/selling dashboards
- [ ] make default demo show "profit" :woozy_face: 
- [ ] have more than 1 warehouses by default for stock dashboard

ankush committed:
chore: use royalty free images

ankush committed:
feat(ux): account workspace number cards

ankush committed:
refactor: gracefully fail while clearing demo data

ankush committed:
fix: confirm before clearing demo data

ankush committed:
chore: telemetry for demo data erasure

deepeshgarg007 committed:
chore: Update customer/supplier names

deepeshgarg007 committed:
fix: Add order data

deepeshgarg007 committed:
Merge branch 'demo_data_on_install' of https://github.com/deepeshgarg007/erpnext into demo_data_on_install
